pkg/controller/import-controller.go

```diff
@@ -366,28 +366,32 @@ func (r *ImportReconciler) updatePvcFromPod(pvc *corev1.PersistentVolumeClaim, p
 
 	log.V(1).Info("Updating PVC from pod")
 	anno := pvc.GetAnnotations()
-	setAnnotationsFromPodWithPrefix(anno, pod, cc.AnnRunningCondition)
 
-	scratchExitCode := false
+	termMsg, err := parseTerminationMessage(pod)
+	if err != nil {
+		log.V(3).Info("Ignoring failure to parse termination message", "error", err.Error())
+	}
+	setAnnotationsFromPodWithPrefix(anno, pod, termMsg, cc.AnnRunningCondition)
+
+	scratchSpaceRequired := termMsg != nil && termMsg.ScratchSpaceRequired != nil && *termMsg.ScratchSpaceRequired
+	if scratchSpaceRequired {
+		log.V(1).Info("Pod requires scratch space, terminating pod, and restarting with scratch space", "pod.Name", pod.Name)
+	}
+
 	if pod.Status.ContainerStatuses != nil &&
-		pod.Status.ContainerStatuses[0].State.Terminated != nil {
-		if message := pod.Status.ContainerStatuses[0].State.Terminated.Message; strings.Contains(message, common.ScratchSpaceRequired) {
-			log.V(1).Info("Pod requires scratch space, terminating pod, and restarting with scratch space", "pod.Name", pod.Name)
-			scratchExitCode = true
-			anno[cc.AnnRequiresScratch] = "true"
```

Comment:
why drop setting this annotation? AFAIK it's needed for things to continue properly

Reply:
Move it to `utils.go` / `setAnnotationsFromPodWithPrefix`, where other annotations are set.

https://github.com/kubevirt/containerized-data-importer/blob/c88d3ba94fd3253e9c31eeb7ceb728f1f95bcf66/pkg/controller/util.go#L305

Reply:
:+1: